builder/common/step_create_ssm_tunnel.go

```diff
@@ -75,15 +84,70 @@ func (s *StepCreateSSMTunnel) Run(ctx context.Context, state multistep.StateBag)
 			LocalPort:  s.LocalPortNumber,
 			Region:     s.Region,
 		}.Start(ssmCtx, ui)
-
 		if err != nil {
 			ui.Error(fmt.Sprintf("ssm error: %s", err))
 		}
 	}()
 
+	var privateKeys [][]byte
+	if len(s.CommunicatorConfig.SSH.SSHPrivateKey) != 0 {
```

Comment:
I'd think we can replace the length check by an empty string comparison for consistency with the `SSHPrivateKeyFile` check below. That is unless I'm misunderstanding the reasons here, if there's a good reason for this implementation we can leave it as-is.
```suggestion
	if s.CommunicatorConfig.SSH.SSHPrivateKey != "" {
```

Reply:
Well I didn't see this before, but it seems that if `SSHPrivateKeyFile` is defined in the configuration, it will set `SSHPrivateKey` to that value (SEE: https://github.com/hashicorp/packer-plugin-amazon/blob/main/builder/common/step_key_pair.go#L34).

Therefore I think we can remove the `if s.CommunicatorConfig.SSH.SSHPrivateKeyFile != ""` block, since it's redundant with this one (in my testing I can confirm that the key is indeed set twice, while not harmful, it's not necessary)

Reply:
Yes exactly, I tested it and the key is set twice. I just wanted to keep this as same of how the [AuthMethod](https://github.dev/hashicorp/packer-plugin-sdk/blob/34ed5d90315ee53e5ebdea1a7c2a68a95b7812b6/communicator/config.go#L394-L395) is set(indeed there it's set twice). But if you agree let's remove it.

Reply:
That's interesting, I suspect this is an unintended side-effect of setting the PrivateKey as the contents of the SSHPrivateKeyFile.
There's probably something to do on that side, in the meantime, we can proceed with removing the `SSHPrivateKeyFile` condition in this code, it doesn't harm to set the public key twice as far as I can tell, but we might as well do it right if we're on it

Reply:
Oh! That makes sense, sorry I didn't catch that, it's good then!